packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -285,18 +290,95 @@ class NestedScrollView extends StatefulWidget {
   }
 
   @override
-  _NestedScrollViewState createState() => _NestedScrollViewState();
+  NestedScrollViewState createState() => NestedScrollViewState();
 }
 
-class _NestedScrollViewState extends State<NestedScrollView> {
+/// The [State] for a [NestedScrollView].
+///
+/// The [ScrollController]s, [innerController] and [outerController], of the
+/// [NestedScrollView]'s children may be accessed through its state. This is
+/// useful for obtaining respective scroll positions in the [NestedScrollView].
+///
+/// If you want to access the inner or outer scroll controller of a
+/// [NestedScrollView], you can get its [NestedScrollViewState] by supplying a
+/// `GlobalKey<NestedScrollViewState>` to the [NestedScrollView.key] parameter).
+/// The outer scroll position can also be obtained directly using
```

Comment:
Let's remove this sentence. Accessing stuff of a widget like that can lead to badness.